### ISSUE TITLE
Categories should be non-negative numbers ERROR

### ISSUE THREAD

**Issue by simonprovost** [automl/auto-sklearn#1203]
Hello everyone, I have been using Auto-Sklearn for about a month and am currently attempting to enter phase 2 of my research when I run into a problem. The error is that no problem will occur with a small amount of time left this task, but I know that after 30 minutes of running/searching in the spatial dimensions of possibilities, it crashed. I am also aware that some issues have been opened/closed regarding it, but none of them have actually aided me even though they presents the same kind of problem, as I have tried all of the suggested solutions without success. The following is the error:

`
raise ValueError('Categories should be non-negative numbers. '
ValueError: Categories should be non-negative numbers. NOTE: floats will be casted to integers.
`

To ensure the type of my columns in my dataset, before any run of the fit() method I am printing the dtype of each of my column and all of them are well "converted" in `category` / `float64` or `int64` which further in the Auto-Sklearn framework will be seen as "Categorical" and "Numerical". However, the problem is still there and I do not know where to go deeper in the framework to find a solution to this problem. Thank you very much if you have anything else to suggest.

1. Using the version of Auto-Sklearn 0.12.6.
2. Using this parameter for the classifier:

```
time_left_for_this_task 180
per_run_time_limit 1
n_jobs 4
memory_limit 5000
seed 85
resampling_strategy holdout
ensemble_size 50
```

3. Using that dataset: https://archive.ics.uci.edu/ml/datasets/Estimation+of+obesity+levels+based+on+eating+habits+and+physical+condition+
4. Splitting (randomly or not it does not matter for the bug I guess) in 10 folds the dataset.
5. Looping over the folds and drop one which will be used as the test_set for later use (not relevant for the bug) and the remaining 9 folds are used for training.
6. 10 classifier are then outputted from the Auto-ML pipeline over the loop but one of them (at the end) crashed. Note: It is even more weirder than it does not crashed at the beginning.

About stracktrace and log file, I do not know where exactly to call 

Please give details about your installation:

* OS: Mac OSX Big sur or Ubuntu 18 LTS.
* On my Mac OS installed or on an Ubuntu virtual machine.
* Python version 3.8.8 for both.
* Auto-sklearn version: 0.12.6.

**Comment by simonprovost** [comment]
Update: 

After re-reading, I missed the point that a fix has been applied during the latest previous release, and indeed, after reading the release log for version 0.13, > [1135](https://github.com/automl/auto-sklearn/pull/1135) Everything appears to work now for folks; I purposefully omitted that comment in issue #1141 first time reading it few days ago, because, as an OSX user, I did not want to restart the installation process, which was already quite lengthy due to all external dependencies with the correct versions to install on our machine. Anyway, I am going to update Auto-Sklearn on both my local and cloud virtual machines and will report back here if it works or not. Given that, the issue can be omitted for the time being, my apologies. 

I will return back here as soon as possible either to close the issue or to report further. Cheers

**Comment by eddiebergman** [comment]
Hi @simonprovost, sorry to hear you had the problem but I'm glad that an update managed to solve your problem. Please let us know if you run into any more issues with this!

**Comment by felidsche** [comment]
Hi,

just in case another OSX user stumbles across this issue.
I've been able to fix it using this simple `Dockerfile`:
```
# auto-sklearn 0.13.0 image
FROM python:3.8
RUN apt-get update -y && apt-get upgrade -y
RUN apt-get install -y curl build-essential swig python3-dev
RUN curl https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt | xargs -n 1 -L 1 pip3 install
RUN pip3 install notebook auto-sklearn
```

and then running:
`docker build -t your_image_tag .`
`docker run -it -v $PWD:/opt/nb -p 8887:8887 --name autosklearn your_image_tag /bin/bash -c "mkdir -p /opt/nb && jupyter notebook --notebook-dir=/opt/nb --ip='0.0.0.0' --port=8887 --no-browser --allow-root"`

Best,
Felix

**Comment by eddiebergman** [comment]
@felidsche thank you for another fix for other OSX users!

Seeing as this was addressed in release 0.13.0 I will close this issue but please feel free to re-use this thread if the problem occurs on any version great 0.13.0

**Comment by simonprovost** [comment]
Everything has been working very well since the update of the 0.13.0 as recommended. I tried over an extensive number of datasets plus hours of run (50 hours in total) and everything is being handled very well. Cheers all!
